outputs/openapi/components.go

```diff
@@ -67,59 +76,80 @@ func makeCollisionSafeNamesFromComponents(components []astra.Field) {
+			bindingTags, uniqueBindings := astra.ExtractBindingTags(component.StructFields)
+			for _, bindingType := range bindingTags {
+				// If sameUntil is greater than 0, we need to remove the package path up to the point where they first don't match
+				if sameUntil > 0 {
+					// Split the package path by "/"
+					splitPackage := strings.Split(component.Package, "/")
+
+					// Pick the final part of the package path, guided by sameUntil
+					// We add 1 because we want to access the first different part of the package path
+					// e.g. github.com/ls6-events/astra and github.com/different/astra would give us sameUntil = 1
+					// and split into "ls6-events" and "different"
+
+					splitPackage = splitPackage[len(splitPackage)-(sameUntil+1):]
+
+					if uniqueBindings {
+						// If there are unique bindings, we need to add the binding type to the collision safe name
+						collisionSafeNames[collisionSafeKey(bindingType, component.Name, component.Package)] = strings.Join(splitPackage, ".") + "." + string(bindingType) + "." + component.Name
+					} else {
+						// If there are no unique bindings, we can just use the package name
+						collisionSafeNames[collisionSafeKey(bindingType, component.Name, component.Package)] = strings.Join(splitPackage, ".") + "." + component.Name
+					}
+				} else {
+					if uniqueBindings {
+						// If there are unique bindings, we need to add the binding type to the collision safe name
+						collisionSafeNames[collisionSafeKey(bindingType, component.Name, component.Package)] = getPackageName(component.Package) + "." + string(bindingType) + "." + component.Name
+					} else {
+						// If there are no unique bindings, we can just use the package name
+						collisionSafeNames[collisionSafeKey(bindingType, component.Name, component.Package)] = getPackageName(component.Package) + "." + component.Name
+					}
+				}
 			}
 		}
 	}
 }
 
 // makeComponentRef creates a reference to the component in the OpenAPI specification
-func makeComponentRef(name, pkg string) string {
-	return "#/components/schemas/" + makeComponentRefName(name, pkg)
+func makeComponentRef(bindingType astTraversal.BindingTagType, name, pkg string) string {
+	return "#/components/schemas/" + makeComponentRefName(bindingType, name, pkg)
 }
 
 // makeComponentRefName converts the component and package name to a valid OpenAPI reference name (to avoid collisions)
-func makeComponentRefName(name, pkg string) string {
-	return collisionSafeNames[collisionSafeKey(name, pkg)]
+func makeComponentRefName(bindingType astTraversal.BindingTagType, name, pkg string) string {
+	return collisionSafeNames[collisionSafeKey(bindingType, name, pkg)]
 }
 
 // componentToSchema converts a component to a schema
-func componentToSchema(component astra.Field) Schema {
-	var schema Schema
-
+func componentToSchema(component astra.Field, bindingType astTraversal.BindingTagType) (schema Schema, bound bool) {
```

Comment:
Nit: personally I don't prefer named return types as it's more difficult for code readability imo

Reply:
Yeah I get what you mean, I just meant I didn't have to initialise two variables, just felt it made the code a bit cleaner but don't feel strongly either way.